feat: add possibility to override touchable ripple color (#3866)
callstack/react-native-paper: docs/docs/guides/11-ripple-effect.md

---
title: Ripple effect
---

> The ripple effect is a visual feedback that occurs when a user interacts with a clickable UI element, such as a button. This response takes the form of a circular ripple expanding from the point of contact, much like a drop falling into water and creating ripples.

Reviewer: ```diff
- clickable
+ pressable
```

> The ripple effect is an essential aspect of Material Design, and Paper's pressable components have it enabled by default. Nonetheless, it can be tailored to suit specific needs.

Reviewer: iOS doesn't have a ripple effect so I think it's worth noting.

## Customize ripple effect color in component

The `rippleColor` prop is available for every pressable component which allows you to set the color of the ripple effect. For the instance, to make the `Button` component's ripple effect transparent, simply pass the desired color value to the prop:

```

> <Button rippleColor="transparent" icon="camera" mode="contained" onPress={() => console.log('Pressed')}>

Reviewer: `transparent`? Wouldn't that be the same as disabling the effect? Let's have a solid color like `#FF0000`.

Author: I've just used the `transparent` color, since based on the issues, users often ask how to customize the color to disable the effect, however, I inclined to your comment and changed the color.

  Press me
</Button>
```

## Disable ripple effect in all components

> To disable the ripple effect in *all* of Paper's components, you need to customize the `Provider`  that is likely wrapping your root component. To do this, simply pass `rippleEffectEnabled: false` to the `settings` object prop within the mentioned `Provider`.

Reviewer: Let's rename `Provider` to `PaperProvider` and add `export const Provider = PaperProvider` and update the docs everywhere to not have `Provider as PaperProvider` and in the next major version we can remove `Provider`.
```suggestion
To disable the ripple effect in **all** of Paper's components set `rippleEffectEnabled: false` on the `settings` prop of `PaperProvider`.
```

Author: > Let's rename Provider to PaperProvider and add export const Provider = PaperProvider and update the docs everywhere to not have Provider as PaperProvider and in the next major version we can remove Provider.

Agree, I will create a separated PR for it.

Author: https://github.com/callstack/react-native-paper/pull/3889
